qa/acceptance-tests/src/test/java/io/camunda/it/client/UsersByTenantSearchTest.java

```diff
@@ -0,0 +1,169 @@
+import java.util.Base64;
+import java.util.List;
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
+
+@MultiDbTest
+@DisabledIfSystemProperty(named = "test.integration.camunda.database.type", matches = "rdbms")
+public class UsersByTenantSearchTest {
+
+  private static final String BASE_PATH = "v2/tenants";
+  private static final String TENANT_ID = Strings.newRandomValidIdentityId();
+  private static final String ADMIN_USER_NAME = "foo";
+  private static final String USER_USERNAME_2 = "testUser1";
+  private static final String USER_USERNAME_3 = "testUser2";
+  private static final String SEARCH_USERNAME = "searchUser";
+  private static final String TEST_PASSWORD = "p455w0rd";
+  @AutoClose private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+
+  @UserDefinition
+  private static final TestUser ADMIN_USER =
+      new TestUser(
+          ADMIN_USER_NAME,
+          TEST_PASSWORD,
+          List.of(
+              new Permissions(ResourceType.AUTHORIZATION, PermissionType.CREATE, List.of("*")),
+              new Permissions(ResourceType.AUTHORIZATION, PermissionType.UPDATE, List.of("*")),
+              new Permissions(ResourceType.USER, PermissionType.CREATE, List.of("*")),
+              new Permissions(ResourceType.TENANT, PermissionType.CREATE, List.of("*")),
+              new Permissions(ResourceType.TENANT, PermissionType.UPDATE, List.of("*"))));
+
+  @UserDefinition
+  private static final TestUser TEST_USER2 =
+      new TestUser(USER_USERNAME_2, TEST_PASSWORD, List.of());
+
+  @UserDefinition
+  private static final TestUser TEST_USER3 =
+      new TestUser(USER_USERNAME_3, TEST_PASSWORD, List.of());
+
+  @UserDefinition
+  private static final TestUser SEARCH_USER =
+      new TestUser(
+          SEARCH_USERNAME,
+          TEST_PASSWORD,
+          List.of(
+              new Permissions(ResourceType.TENANT, PermissionType.READ, List.of("*")),
+              new Permissions(ResourceType.TENANT, PermissionType.UPDATE, List.of("*")),
+              new Permissions(ResourceType.APPLICATION, PermissionType.ACCESS, List.of("*")),
+              new Permissions(ResourceType.RESOURCE, PermissionType.READ, List.of("*"))));
+
+  @BeforeAll
+  static void setup(
+      @Authenticated(ADMIN_USER_NAME) final CamundaClient adminClient,
+      @Authenticated(USER_USERNAME_2) final CamundaClient user2Client,
+      @Authenticated(USER_USERNAME_3) final CamundaClient user3Client) {
+
+    createTenant(adminClient, TENANT_ID);
+    assignUserToTenant(adminClient, ADMIN_USER_NAME, TENANT_ID);
+    assignUserToTenant(adminClient, USER_USERNAME_2, TENANT_ID);
+    assignUserToTenant(adminClient, USER_USERNAME_3, TENANT_ID);
+
+    waitUntilExactUsersExist(
+        adminClient, ADMIN_USER_NAME, TENANT_ID, ADMIN_USER_NAME, USER_USERNAME_2, USER_USERNAME_3);
```

Comment:
@nicpuppa the test should be fixed now. The issue was that the user assignment wasn't written to secondary storage before the test case. I added an assertion that will block until the users can be read from secondary storage by an "admin user".